environments/common/ecr/ecr.tf

```
@@ -26,6 +25,36 @@ resource "aws_ecr_repository" "this" {
   }
 }
 
+resource "aws_ecr_lifecycle_policy" "expire_untagged_images_policy" {
+  for_each = toset(local.applications)
+
+  repository = "tariff-${each.key}"
+
+  policy = <<EOF
+{
+    "rules": [
+        {
+            "rulePriority": 1,
+            "description": "Expire untagged images older than 14 days",
+            "selection": {
+                "tagStatus": "untagged",
+                "countType": "sinceImagePushed",
+                "countUnit": "days",
+                "countNumber": ${lookup(each.value, "expire_untagged_images_days", 14)}
+            },
+            "action": {
+                "type": "expire"
+            }
+        }
+    ]
+}
+EOF
```

```suggestion
  policy = jsonencode({
    rules = [{
      rulePriority = 1
      description = "Expire untagged images older than 14 days"
      selection = {
        tagStatus   = "untagged"
        countType   = "sinceImagePushed"
        countUnit   = "Days"
        countNumber = lookup(each.value, "expire_untagged_images_days"
      }
      action = {
        type = "expire"
      }
    }]
  })
```

Could we use a `jsonencode()` here rather than a heredoc? That's more consistent with the rest of the repo 😄

---

sure, I just borrowed it from standard terraform docs